src/test/outmdsd/MockServer.cc

```diff
@@ -197,6 +197,30 @@ MockServer::RunEchoTask(
     m_connfdSet.insert(connfd);
     lck.unlock();
 
+    // Because RunEchoTask is called in an async task, it is better to handle
+    // exceptions at task level. Otherwise, the exceptions won't show up in
+    // master thread until task.get() is called, which could cause unexpected delays
+    // at handling client communications.
+    try {
+        StartReadLoop(connfd);
+    }
+    catch(const std::exception& ex) {
+        Log(std::string("Error: RunEchoTask() failed: ") + ex.what());
+
+        std::unique_lock<std::mutex> lck(m_connMutex);
+        m_connfdSet.erase(connfd);
+        lck.unlock();
+
+        shutdown(connfd, SHUT_RDWR);
+        close(connfd);
```

Comment:
the master thread doesn't care/catch these exceptions. it just starts async tasks. and wait for them to finish before it close. the right place is to handle the exception at task level and log errors if failed. if a test client needs to continue talking with MockServer, client should re-initiate socket/connection().

---
In reply to: [96991364](https://github.com/Azure/fluentd-plugin-mdsd/pull/3#discussion_r96991364) [](ancestors = 96991364)